test/functions/promotion/racy-arg.chpl

```diff
@@ -0,0 +1,62 @@
+proc increment(ref rcv: int, x: int) {
+  rcv += x;
+}
+
+proc noop(const ref rcv: int, x: int) {
+}
+
+config const n = 5000;
+config const verbose = false;
+
+proc test1() {
+  var x = 42;
+  noop(x, 1);
+  increment(x, 1); // works as expected
+  writeln("1A expect ", 43);
+  writeln("1A    got ", 43);
+
+  var A: [1..n] int = 1..n;
+  var B: [1..n] int;
+  noop(B, A);
+  increment(B, A); // works as expected
+  writeln("1B expect ", + reduce A);
+  writeln("1B    got ", + reduce B);
+  if verbose then
+    writeln(B);
+
+  noop(x, A);
+  increment(x, A); // "works" but should arguably generate an error
```

Comment:
This comment is out of date, right?

Reply:
Or is that to indicate that it would work without the current error?

Reply:
I'll update the comment